cpp/oneapi/dal/algo/jaccard/common.hpp

```diff
@@ -52,19 +68,28 @@ class ONEDAL_EXPORT descriptor_base : public base {
     void set_block_impl(const std::initializer_list<std::int64_t>& row_range,
                         const std::initializer_list<std::int64_t>& column_range);
 
-    dal::detail::pimpl<detail::descriptor_impl> impl_;
+    dal::detail::pimpl<detail::descriptor_impl<task_t>> impl_;
 };
 
+} // namespace detail
+
 /// Class for the Jaccard similarity algorithm descriptor
 ///
 /// @tparam Float The data type of the result
 /// @tparam Method The algorithm method
-template <typename Float = float, typename Method = method::by_default>
-class descriptor : public descriptor_base {
-    using base_t = descriptor_base;
+template <typename Float = float,
+          typename Method = method::by_default,
+          typename Task = task::by_default>
+class descriptor : public detail::descriptor_base<Task> {
+    static_assert(detail::is_valid_method<Method>);
+    static_assert(detail::is_valid_task<Task>);
+
+    using base_t = detail::descriptor_base<Task>;
 
 public:
+    using float_t = Float;
     using method_t = Method;
+    using task_t = Task;
 
     /// Creates a new instance of the class with the default property values.
```

Comment:
do we use the same terminology for constructor across the classes?

Reply:
Yes